core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -154,12 +232,93 @@ oppia.directive('questionPlayer', [
             return ctrl.totalQuestions;
           };
 
+          var createScorePerSkillMapping = function() {
+            var scorePerSkillMapping = {};
+            if (questionPlayerConfig.skillList) {
+              for (var i = 0; i < questionPlayerConfig.skillList.length; i++) {
+                var skillId = questionPlayerConfig.skillList[i];
+                var description = questionPlayerConfig.skillDescriptions[i];
+                scorePerSkillMapping[skillId] = {
+                  description: description,
+                  score: 0.0
+                };
+              }
+            }
+            return scorePerSkillMapping;
+          };
+
+          var calculateScores = function(questionStateData) {
+            var scorePerSkillMapping = createScorePerSkillMapping();
+            $scope.resultsLoaded = false;
+            var totalScore = 0.0;
+            var minScore = Number.MAX_VALUE;
+            var totalQuestions = Object.keys(questionStateData).length;
+            for (var question in questionStateData) {
+              var questionData = questionStateData[question];
+              var totalHintsPenalty = 0.0;
+              var wrongAnswerPenalty = 0.0;
+              if (questionData.answers) {
+                wrongAnswerPenalty = (
+                  (questionData.answers.length - 1) * WRONG_ANSWER_PENALTY);
+              }
+              if (questionData.usedHints) {
+                totalHintsPenalty = (
+                  questionData.usedHints.length * VIEW_HINT_PENALTY);
+              }
+              var questionScore = 1.0;
+              if (questionData.viewedSolution) {
+                questionScore = 0.0;
+              } else {
+                // If questionScore goes negative, set it to 0
+                questionScore = Math.max(
+                  0, questionScore - totalHintsPenalty - wrongAnswerPenalty);
+              }
+              // Calculate total score
+              ctrl.totalScore += questionScore;
+
+              // Calculate scores per skill
+              if (!(questionData.linkedSkillIds)) {
+                continue;
+              }
+              for (var i = 0; i < questionData.linkedSkillIds.length; i++) {
+                var skillId = questionData.linkedSkillIds[i];
+                if (!(skillId in scorePerSkillMapping)) {
+                  continue;
+                }
+                scorePerSkillMapping[skillId].score += questionScore;
```

Comment:
Isn't this stored somewhere in the backend?

Reply:
Sorry I didn't completely understand your question - Are you asking if we store these scores in the backend? If so then no as of now we won't be. @sophiewu6 's GSoC project will be handling that bit

Reply:
Oh thanks for that. I wasn't sure of the scope of this PR. Yeah, that was my intention of this comment. I assumed that the score should be posted to the backend to be used to track progress/levels or something like that.